diff --git a/include/cutlass/layout/tensor.h b/include/cutlass/layout/tensor.h
index 9e8a354e..6d32dc9a 100644
--- a/include/cutlass/layout/tensor.h
+++ b/include/cutlass/layout/tensor.h
@@ -210,7 +210,7 @@ public:
         || (extent.h() * stride_[1] > stride_[2])) {
       assert(0);
     }
-    return extent.n() * stride_[2];
+    return LongIndex(extent.n()) * LongIndex(stride_[2]);
   }
 };
 
@@ -290,7 +290,7 @@ public:
   /// Compute the number of contiguous elements needed to store a tensor with the given size
   CUTLASS_HOST_DEVICE
   LongIndex capacity(TensorCoord const &extent) const {
-    return extent.n() * stride_[2];
+    return LongIndex(extent.n()) * LongIndex(stride_[2]);
   }
 };
 
@@ -634,7 +634,7 @@ public:
         || (extent.d() * stride_[2] > stride_[3])) {
       assert(0);
     }
-    return extent.n() * stride_[3];
+    return LongIndex(extent.n()) * LongIndex(stride_[3]);
   }
 };
 
diff --git a/include/cutlass/pipeline/sm90_pipeline.hpp b/include/cutlass/pipeline/sm90_pipeline.hpp
index aae17d98..46f0747c 100644
--- a/include/cutlass/pipeline/sm90_pipeline.hpp
+++ b/include/cutlass/pipeline/sm90_pipeline.hpp
@@ -491,7 +491,7 @@ public:
     consumer_release(state.index());
   }
 
-private:
+public:
   uint32_t dst_blockid_ = 0;
   uint32_t is_signaling_thread_ = 0;
   FullBarrier *full_barrier_ptr_ = nullptr;
@@ -915,7 +915,7 @@ public:
     consumer_release(state.index());
   }
 
-private:
+public:
   FullBarrier *full_barrier_ptr_ = nullptr;
   EmptyBarrier *empty_barrier_ptr_ = nullptr;
   Params params_;
